examples/secured-client-with-oauth2/secured_client_with_oauth2.bal

```diff
@@ -2,30 +2,101 @@ import ballerina/http;
 import ballerina/log;
 
 // Define the OAuth2 client endpoint to call the backend services.
-// OAuth2 authentication is enabled by setting the `scheme: http:OAUTH2`
-// If `accessToken` is invalid, it will be automatically refreshed with the
-// provided `clientId`, `clientSecret`, `refreshToken`, `refreshUrl`.
-http:Client httpEndpoint = new("https://www.googleapis.com/tasks/v1", config = {
-    auth: {
-        scheme: http:OAUTH2,
-        config: {
-            accessToken: "ya29.GlvQBkqJS0yn0zsZm4IIUUzLk3DH1rRiCMKnHiz6deycKmTFiDsuoFlFfrmXF8dCb0gyzLyXpnv3VcrIlauj3nMs61CbydaAqMl6RwVIU2r2qg1StVVvxRWT9_Or",
-            clientId: "506144513496-dqm5vdqfrfhdjjom10rmvafb8e3h7rtm.apps.googleusercontent.com",
-            clientSecret: "3hw2XN4MfiIRrv6mghX6m5gM",
-            refreshToken: "1/UwH3YyYccKTrH9bqj35Y7hMYTK9f3HEC3uzlrleFwPE",
-            refreshUrl: "https://www.googleapis.com/oauth2/v4/token"
+// Setting the `scheme: http:OAUTH2` enables OAuth2 authentication.
+// Setting the `grantType: http:CLIENT_CREDENTIALS_GRANT` configures the
+// OAuth2 client credentials grant type.
+http:Client clientEP1 = new("https://api.bitbucket.org/2.0", config = {
+        auth: {
+            scheme: http:OAUTH2,
+            config: {
+                grantType: http:CLIENT_CREDENTIALS_GRANT,
+                config: {
+                    tokenUrl: "https://bitbucket.org/site/oauth2/access_token",
+                    clientId: "mMNWS9PLmM93V5WHjC",
+                    clientSecret: "jLY6xPY3ER4bNTspaGu6fb7kahhs7kUa"
+                }
+            }
         }
-    }
-});
+    });
+
+// Define the OAuth2 client endpoint to call the backend services.
+// Setting the `scheme: http:OAUTH2` to enable OAuth2 authentication.
+// Setting the `grantType: http:PASSWORD_GRANT` configures the
+// OAuth2 password grant type.
+// If the access token expires or become invalid, then it will be automatically
+// refreshed with the provided `refreshUrl`.
+http:Client clientEP2 = new("https://api.bitbucket.org/2.0", config = {
+        auth: {
+            scheme: http:OAUTH2,
+            config: {
+                grantType: http:PASSWORD_GRANT,
+                config: {
+                    tokenUrl: "https://bitbucket.org/site/oauth2/access_token",
+                    username: "b7a.demo@gmail.com",
+                    password: "39^R3ODx",
```

Comment:
Let's use a password showing that the value we have here is just a dummy value. Change the actual password in bitbucket (since github maintain code history), and test for negative condition for this client.